src/main.c

```diff
@@ -3176,9 +3207,14 @@ int main(int argc, char **argv)
         error_log_fd = -1;
     }
 
-    /* start the threads */
+
     conf.threads = alloca(sizeof(conf.threads[0]) * conf.thread_map.size);
     size_t i;
+    for (i = 0; i < conf.thread_map.size; i++) {
+        h2o_context_init(&conf.threads[i].ctx, h2o_evloop_create(), &conf.globalconf);
+    }
```

Comment:
I believe I had to move this to before we enter sandbox mode. The mruby code requires a file descriptor, and it's not clear to me if we want to have mruby use the privileged process to acquire the file descriptor, or move the mruby initialization to before the sandbox entrance. I used to have the sandbox pin in the threads, but this isnt portable since FreeBSD drops the entire process into a sandbox (all threads) vs seccomp and Linux pins the policies on a per thread basis. I think I was hitting a condition where the first thread dropped the entire process into the sandbox, and subsequent initialization operations were failing. Moving this to explicitly be outside of the sandbox was a way to work around it. But I would be open to feedback

Reply:
I think it might be a good idea to extend and use startup_sync_barrier. It is a barrier used to synchronize the moment when the threads start accepting connections / packets.

Maybe we can add an argument to `h2o_barrier_init` for passing a callback to be executed when the barrier is about to exit, then use that callback to invoke `h2o_priv_bind_sandbox`.